azkaban-common/src/test/java/azkaban/executor/ExecutorApiGatewayTest.java

```diff
@@ -85,23 +90,100 @@ public void updateExecutions() throws Exception {
   }
 
   @Test
-  public void testPathWithDefaultConfigs() {
+  public void testPathWithDefaultConfigs() throws Exception {
     int executionId = 12345;
-    String path = gateway.createExecutionPath(12345);
+    String path = gateway.createExecutionPath(Optional.of(12345));
     Assert.assertEquals("/"+ExecutorApiGateway.DEFAULT_EXECUTION_RESOURCE, path);
   }
 
   @Test
-  public void testPathWithContainerizedConfigs() {
-    Props containerizedConfigs = new Props();
-    containerizedConfigs.put(ConfigurationKeys.AZKABAN_EXECUTOR_REVERSE_PROXY_ENABLED, "true");
-    containerizedConfigs.put(ConfigurationKeys.AZKABAN_EXECUTION_DISPATCH_METHOD,
-        DispatchMethod.CONTAINERIZED.name());
-    ExecutorApiGateway gateway = new ExecutorApiGateway(client, containerizedConfigs);
+  public void testPathWithContainerizedConfigs() throws Exception {
+    final ExecutorApiGateway gateway = gatewayWithConfigs(this.client, containerizationEnabledProps());
     int executionId = 12345;
-    String path = gateway.createExecutionPath(12345);
+    String path = gateway.createExecutionPath(Optional.of(12345));
     Assert.assertEquals("/" + executionId + "/" +
             ExecutorApiGateway.CONTAINERIZED_EXECUTION_RESOURCE,
         path);
   }
+
+  private Props containerizationEnabledProps() {
+    Props containerizedProps = new Props();
+    containerizedProps.put(ConfigurationKeys.AZKABAN_EXECUTOR_REVERSE_PROXY_ENABLED, "true");
+    containerizedProps.put(AZKABAN_EXECUTOR_REVERSE_PROXY_HOSTNAME, REVERSE_PROXY_HOST);
+    containerizedProps.put(AZKABAN_EXECUTOR_REVERSE_PROXY_PORT, REVERSE_PROXY_PORT);
+    containerizedProps.put(ConfigurationKeys.AZKABAN_EXECUTION_DISPATCH_METHOD,
+        DispatchMethod.CONTAINERIZED.name());
+    return containerizedProps;
+  }
+
+  private ExecutorApiGateway gatewayWithConfigs(ExecutorApiClient client, Props props) {
+    return new ExecutorApiGateway(client, props);
+  }
+
+  @Test
+  public void testPingWithDefaultConfigs() throws Exception {
+    Props props = new Props();
+    final ExecutorApiClient clientSpy = Mockito.spy(new SendDisabledExecutorApiClient(props));
+
+    final ExecutorApiGateway gateway = gatewayWithConfigs(clientSpy, props);
+    final String apiAction = ConnectorParams.PING_ACTION;
+    final String apiHost = "host1";
+    final int apiPort = 1234;
+    final Map<String, Object> response = gateway.callWithExecutionId(apiHost, apiPort, apiAction,
```

Comment:
Please write a comment here, that this call is not expected to throw an exception something like comment at line 171